modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/file/FileStorage.kt

```diff
@@ -0,0 +1,295 @@
+                        moveFileToDir(downloadedFilePath, episodesDir)?.let { updatedPath ->
+                            episodesManager.updateDownloadFilePath(episode, updatedPath, markAsDownloaded = false)
+                        }
+                    }
+                }
+
+                val oldCustomFilesDir = getOrCreateDir(oldPocketCastsDir, DIR_CUSTOM_EPISODES)
+                val newCustomFilesDir = getOrCreateDir(newPocketCastsDir, DIR_CUSTOM_EPISODES)
+                if (oldCustomFilesDir.exists()) {
+                    moveDir(oldCustomFilesDir, newCustomFilesDir)
+                }
+
+                val oldNetworkImageDir = getOrCreateDir(oldPocketCastsDir, DIR_NETWORK_IMAGES)
+                val newNetworkImageDir = getOrCreateDir(newPocketCastsDir, DIR_NETWORK_IMAGES)
+                if (newNetworkImageDir.exists()) {
+                    moveDir(oldNetworkImageDir, newNetworkImageDir)
+                }
+            } else {
+                LogBuffer.e(LogBuffer.TAG_BACKGROUND_TASKS, "Old directory did not exist")
+            }
+        } catch (e: StorageException) {
+            LogBuffer.e(LogBuffer.TAG_BACKGROUND_TASKS, e, "Unable to move storage to new location")
+        }
+    }
+
+    fun fixBrokenFiles(episodeManager: EpisodeManager) {
+        try {
+            // Get all possible locations
+            val dirPaths = buildSet {
+                addAll(StorageOptions().getFolderLocations(context).map(FolderLocation::filePath))
+                add(context.filesDir.absolutePath)
+                val customDir = settings.getStorageCustomFolder()
+                if (customDir.isNotBlank() && File(customDir).exists()) {
+                    add(customDir)
+                }
+            }
+
+            // Search each directory for missing files
+            dirPaths.forEach dirIteration@{ dirPath ->
+                val dir = File(dirPath)
+                if (!dir.exists() || !dir.canRead()) {
+                    return@dirIteration
+                }
+                val pocketCastsDir = File(dir, "PocketCasts")
+                if (!pocketCastsDir.exists() || !pocketCastsDir.canRead()) {
+                    return@dirIteration
+                }
+                val episodesDir = File(pocketCastsDir, DIR_EPISODES)
+                if (!episodesDir.exists() || !episodesDir.canRead()) {
+                    return@dirIteration
+                }
+                episodesDir.listFiles()?.forEach fileIteration@{ file ->
+                    val fileName = file.name
+                    val dotPosition = fileName.lastIndexOf('.')
+                    if (dotPosition < 1) {
+                        return@fileIteration
+                    }
+                    val uuid = fileName.substring(0, dotPosition)
+                    if (uuid.length != 36) {
+                        return@fileIteration
+                    }
+
+                    @Suppress("DEPRECATION")
```

Comment:
Should we remove deprecated code?

Reply:
Yes, but I didn't want to introduce such changes in this PR to reduce the review overhead only to pure code migration. Maybe with some minor improvements. Removing this deprecation requires introducing a suspending modifier to the function. I have a separate [branch](https://github.com/Automattic/pocket-casts-android/tree/task/file-storage-kotlin-improvements) with such improvements.

Reply:
> I didn't want to introduce such changes in this PR to reduce the review overhead only to pure code migration.

Thank you for this! 🙇 🙂